src/components/top-bar/top-bar.tsx

```diff
@@ -85,22 +86,26 @@ const renderBreadcrumbs = (breadcrumbs: { title: string | React.ReactNode, path?
     </div>
 );
 
-const renderActionMenu = (actionMenu: ActionMenu) => (
+const renderActionMenu = (actionMenu: ActionMenu, actionMenuExtensions: React.ReactNode[]) => (
     <div>
         {actionMenu.items.map((item, i) => (
             <button disabled={!!item.disabled} qe-id={item.qeId} className='argo-button argo-button--base' onClick={() => item.action()} style={{marginRight: 2}} key={i}>
                 {item.iconClassName && (<i className={item.iconClassName} style={{marginLeft: '-5px', marginRight: '5px'}}/>)}
                 {item.title}
             </button>
         ))}
+        {actionMenuExtensions && actionMenuExtensions.map((ext, index) => (
```

Comment:
Adding another button does not require an `argo-ui` change, you can just add another `item` to the list in CD.

Reply:
I still want to pursue these changes, and I think this is an easy solution that does not require any extensions or argo. If we have a better solution in the future, I will be happy to review it.

Reply:
> I think this is an easy solution that does not require any extensions or argo

It's definitely not "easy" given how invasive it is, and it adds extensions to `argo-ui`, which does in fact add logic to Argo

Reply:
See the concrete guidance I left downstream for how to avoid a change in `argo-ui`: https://github.com/argoproj/argo-cd/pull/19620/files/3e67e6dbda7c7f716a35164c636ad174df0e7396#r1727992070